File: Algorithms/C/11 Container With Most Water.c
int max(int a, int b)
{
    return a > b ? a : b;
}

int min(int a, int b)
{
    return a < b ? a : b;
}

int maxArea(int* height, int heightSize){
    int area = 0;
    int left, right;
    left = 0;
    right = heightSize - 1;
    while (left < right) {
        area = max(area, min(height[left], height[right]) * (right - left));
        if (height[left] < height[right]) {
            ++left;
        }
        else {
            --right;
        }
    }
    return area;
}

File: Algorithms/Python/11 Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = 0
        left, right = 0, len(height) - 1
        while left < right:
            area = max(area, min(height[left], height[right]) * (right - left))
            if height[left] < height[right]:
                left = left + 1
            else:
                right = right - 1
        return area
